命令行关闭节点 p2p panic

cli close 关闭节点，p2p panic


panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0xf49d10]

goroutine 305 [running]:
github.com/33cn/chain33/system/p2p/dht/protocol/peer.(*Protocol).refreshSelf(0xc003400700)
        /home/jiangpeng/go/src/github.com/33cn/chain33/system/p2p/dht/protocol/peer/peerinfo.go:60 +0x30
github.com/33cn/chain33/system/p2p/dht/protocol/peer.InitProtocol.func1(0xc003400700)
        /home/jiangpeng/go/src/github.com/33cn/chain33/system/p2p/dht/protocol/peer/peer.go:101 +0x170
created by github.com/33cn/chain33/system/p2p/dht/protocol/peer.InitProtocol
        /home/jiangpeng/go/src/github.com/33cn/chain33/system/p2p/dht/protocol/peer/peer.go:88 +0x607



peer info相关代码 @libangzhu

--- a/system/p2p/dht/protocol/peer/handler.go
+++ b/system/p2p/dht/protocol/peer/handler.go
@@ -13,6 +13,9 @@ import (
 
 func (p *Protocol) handleStreamPeerInfo(stream network.Stream) {
 	peerInfo := p.getLocalPeerInfo()
+	if peerInfo == nil {
+		return
+	}
 	err := protocol.WriteStream(peerInfo, stream)
 	if err != nil {
 		log.Error("handleStreamPeerInfo", "WriteStream error", err)
@@ -64,6 +67,9 @@ func (p *Protocol) handleStreamPeerInfoOld(stream network.Stream) {
 	}
 
 	peerInfo := p.getLocalPeerInfo()
+	if peerInfo == nil {
+		return
+	}
 	pInfo := &types.P2PPeerInfo{
 		Addr:           peerInfo.Addr,
 		Port:           peerInfo.Port,

